src/encoded/static/components/matrix_entex.js

```diff
@@ -226,59 +273,114 @@ const convertContextToDataTable = (context) => {
     const cellSubcategory = context.matrix.x.group_by[3];
     const rowCategory = context.matrix.y.group_by[0];
     const rowSubcategory = context.matrix.y.group_by[1];
+    const rowKeys = [];
+    const headerRows = [];
 
     // Generate the mapping of column categories and subcategories.
-    const colMap = generateColMap(context);
+    const { colMap, targetAssays } = generateColMap(context);
     const colCount = Object.keys(colMap).length;
 
     // Convert column map to an array of column map values sorted by column number for displaying
     // in the matrix header.
     const sortedCols = Object.keys(colMap).map(assayColKey => colMap[assayColKey]).sort((colInfoA, colInfoB) => colInfoA.col - colInfoB.col);
 
-    // Generate array of names of assays that have targets and don't collapse their targets, for
-    // rendering those columns as disabled.
-    const colCategoriesWithSubcategories = Object.keys(colMap).filter(colCategoryName => colMap[colCategoryName].hasSubcategories && !collapsedAssays.includes(colCategoryName));
+    // Generate the matrix header row labels for the assays with targets. Need a max-width inline
+    // style so that wide labels don't make the target columns expand.
+    let targetAssayHeader;
```

Comment:
Should targetAssayHeader be in the if-statement as it seems not to be used anywhere else. If yes, then it can be const and be just a tad faster than using let in evergreen browsers.

Reply:
I’d like to address this one. I think it might need to be a const, but I need to see it in more context after I’m done with one thing first.

Reply:
Yes, you’re right. I pushed a change to make it a const, and updated the demo: https://encd-5053-entex-update-pr2-fytanaka.demo.encodedcc.org/entex-matrix/?type=Experiment&status=released&internal_tags=ENTEx